mleap-spark-extension/build.sbt

```diff
@@ -2,3 +2,18 @@ import ml.combust.mleap.{Dependencies, Common}
 
 Common.defaultMleapSettings
 Dependencies.sparkExtension
+
+val writeRuntimeClasspathToFile = taskKey[Unit]("Writes runtime classpath to a file")
+
+writeRuntimeClasspathToFile := {
```

Comment:
Sorry for my late answer to question 3! I had to look into this.

If we need to pass the classpath to python tests, have you considered to pass an environment variable ? Execution would look like:
```
MLEAP_CLASSPATH='/some/folder/with/classpath/' tox -e py36
```

In order to generate the classpath, I think there are some useful sbt commands, this [Stackoverflow answer](https://stackoverflow.com/a/21738753) shows some useful ones.

Reply:
My goal was this: export classpath to a file as part of any regular sbt compile that affects mleap-spark-extension.

Having sbt task that does it is the first step – check.

Problem 1: I failed to find a way to make any compile (regular compile vs. test:compile) include this task. For example `sbt "inspect tree mleap-spark-extension/test:compile"` showed that it doesn't depend on `writeRuntimeClasspathToFile`.

Travis is instructed to run `sbt test`. I don't really know how this works with sbt, but `test:compile` also compiles the runtime sources, not only test sources. However doesn't execute the regular `compile:compile` at all and so making that depend on `writeRuntimeClasspathToFile` won't make `sbt test` depend on it.

----

Otherwise this produces the same output as my custom `writeRuntimeClasspathToFile` task:

`sbt --error "export mleap-spark-extension/runtime:fullClasspath" > fullClasspath.txt`

(`--error` is needed to disable info logging printed into the the file as well)

Problem 2: I couldn't find how to run the same "command" as scala from sbt file, but I guess it should be doable. That would allow simplifying the "impl" of the writeRuntimeClasspathToFile task.

Reply:
> have you considered to pass an environment variable

I didn't. If we go there, I would prefer it to be done in a IDE-compatible way, not something that is set in the `Makefile` or some travis conf file and only works when executing tests through them.

If switching to pytest, I know a clean way at least:
- set the env var in `conftest.py` with `os.environ` in `def pytest_generate_tests(metafunc)`
- that way env var is available always when running `pytest` (IDE or command line) without any manual setup

In general I would like this PR's scope to remain small enough, and then improve later with switching to pytest, if anyone is up for that.

Reply:
Aren't env. variables IDE-compatible? they can generally be set in the configuration section and then used in the testing environment.

How's the file cleaned up? What if a testing machine has write-permissions disabled or if I run sbt from a different folder?  I'd rather have a sbt task printing `dependencyClasspath in Runtime` to screen and then do things with it.

Also happy to use pytest if possible.

Reply:
> Aren't env. variables IDE-compatible? they can generally be set in the configuration section and then used in the testing environment.

Yes, it can be done, but is a manual step. I'd really like this to work out of the box.

> How's the file cleaned up?

It doesn't need to be cleaned up, but if I'm not mistaken it would be cleaned up as part of `sbt clean`. Once again, there's an analogy to the .class files from build output.

> What if a testing machine has write-permissions disabled

I don't see how it could then compile the classes.

> I run sbt from a different folder

I don't think sbt can be run from any other folder than root. Or can it? If yes, it should be easy to get the module's build output folder in build.sbt, so that the path for classpath-runtime.txt can be relative to that.

> I'd rather have a sbt task printing dependencyClasspath in Runtime to screen and then do things with it.

I really don't understand why that would be better. Sorry :/ To me it just seems like redundant manual steps to run the tests.

> Also happy to use pytest if possible.

Definitely, but not part of this PR, please. AFAIK pytest doesn't make it easier to have the classpath available for the tests. It's more about invoking sbt.

Reply:
> My goal was this: export classpath to a file as part of any regular sbt compile that affects mleap-spark-extension.

> Problem 1: I failed to find a way to make any compile (regular compile vs. test:compile) include this task

I found a way by using sbt's triggeredBy. Committed it here: https://github.com/combust/mleap/pull/571/commits/4a0ce0738ab19ca45ca459d6686d689cd54321f8